sdk/core/core-client-rest/review/core-client.api.md

```diff
@@ -138,11 +138,17 @@ export interface OperationRequestOptions {
 
 // @public
 export type PathParameters<TRoute extends string> = TRoute extends `${infer _Head}/{${infer _Param}}${infer Tail}` ? [
-pathParameter: string,
+pathParameter: string | PathParameterWithOptions,
 ...pathParameters: PathParameters<Tail>
 ] : [
 ];
 
+// @public
+export interface PathParameterWithOptions {
+    allowReserved?: boolean;
```

Comment:
Seems reasonable to me -- @MaryGao what do you think? Is it also worth considering `skipUrlEncoding`, for consistency, since that's the name of the request-level parameter which has similar effect?

Reply:
I agree more obvious is better. But here we will not re-invent a new word and we just leverage words in [OpenAPI 3](https://swagger.io/docs/specification/serialization/) e.g allowReserved/explode/style and they would finally expose to customers in RLC. So I prefer keeping it as it's. If these words are the same or similar meaning with OpenAPI 3 then customers could easily touch relevant references by Google.

Reply:
Sounds reasonable too.